packages/cli/src/project.ts

```diff
@@ -436,6 +436,22 @@ export class Project {
     return forceRecompile || changedSources.includes(name)
   }
 
+  private async checkMethodIndex(newArtifact: Compiled<Contract>) {
+    const artifactPath = newArtifact.sourceInfo.getArtifactPath(this.artifactsRootDir)
+    let oldArtifact: Contract
+    try {
+      oldArtifact = await Contract.fromArtifactFile(artifactPath, '', '')
+    } catch (error) {
+      throw new Error(`Failed to load contract artifact, error: ${error}, contract: ${newArtifact.sourceInfo.name}`)
+    }
+    newArtifact.artifact.functions.forEach((newFuncSig, index) => {
+      const oldFuncSig = oldArtifact.functions[`${index}`]
+      if (oldFuncSig.name !== newFuncSig.name) {
+        throw new Error(`Function ${newFuncSig.name} has invalid index ${index}`)
```

Comment:
`The newly compiled contract ${newfuncSig.name} has different method indexes compared to the existing compilation.`